index.d.ts

```diff
@@ -0,0 +1,27 @@
+// TypeScript typings for "mocha-testdata@1.1.3".
+
+interface MochaGiven {
+    <P1, P2, P3>(...params: { 0: P1, 1: P2, 2: P3 }[]): MochaGivenMethods3<P1, P2, P3>;
+    <P1, P2>(...params: { 0: P1, 1: P2 }[]): MochaGivenMethods2<P1, P2>;
+    <P1>(...params: P1[]): MochaGivenMethods1<P1>;
```

Comment:
mocha-testdata, unfortunately, treats a _single_ array argument as a `params` argument, so I think this interface is incomplete. I think we need one more set of overloads:

```ts
interface MochaGiven {
    <P1, P2, P3>(...params: { 0: P1, 1: P2, 2: P3 }[]): MochaGivenMethods3<P1, P2, P3>;
    <P1, P2>(...params: { 0: P1, 1: P2 }[]): MochaGivenMethods2<P1, P2>;
    <P1>(...params: P1[]): MochaGivenMethods1<P1>;

    <P1, P2, P3>(params: { 0: P1, 1: P2, 2: P3 }[]): MochaGivenMethods3<P1, P2, P3>;
    <P1, P2>(params: { 0: P1, 1: P2 }[]): MochaGivenMethods2<P1, P2>;
    <P1>(params: P1[]): MochaGivenMethods1<P1>;
}
```